### PR TITLE
Temporary channel for data transfer

### PR DESCRIPTION
**Pull request by hongdami**
Signed-off-by: Miao, Hongda <hongda.miao@intel.com>

Issue Number: 

Objective of pull request:

## Pull request checklist

Your PR fulfills the following requirements:
- [ ] [Issue](https://github.com/lava-nc/lava/issues) created that explains the change and why it's needed
- [ ] Tests are part of the PR (for bug fixes / features)
- [ ] [Docs](https://github.com/lava-nc/docs) reviewed and added / updated if needed (for bug fixes / features)
- [ ] PR conforms to [Coding Conventions](https://lava-nc.org/developer_guide.html#coding-conventions)
- [ ] [PR applys BSD 3-clause or LGPL2.1+ Licenses](https://lava-nc.org/developer_guide.html#add-a-license) to all code files
- [ ] Lint (`flakeheaven lint src/lava tests/`) and (`bandit -r src/lava/.`) pass locally
- [ ] Build tests (`pytest`) passes locally

## Pull request type

 

 

Please check your PR type:

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation changes
- [ ] Other (please describe): 

## What is the current behavior?

-

## What is the new behavior?

-

## Does this introduce a breaking change?

- [ ] Yes
- [ ] No

### CODE REVIEW

**Review by killight98 — CHANGES_REQUESTED** [review]

File: src/lava/magma/runtime/message_infrastructure/message_infrastructure/csrc/channel/socket/socket.h [hunk 1]
```diff
@@ -35,13 +39,16 @@ class SktManager {
   SktManager& operator=(SktManager&&) = delete;
 
   SocketPair AllocChannelSocket(size_t nbytes);
+  SocketFile AllocSocketFile(const std::string addr_path);
```
Comment (killight98) [inline src/lava/magma/runtime/message_infrastructure/message_infrastructure/csrc/channel/socket/socket.h:42]:
```
SocketFile AllocSocketFile(const std::string &addr_path);
```

File: src/lava/magma/runtime/message_infrastructure/message_infrastructure/csrc/channel/socket/socket.h [hunk 2]
```diff
@@ -35,13 +39,16 @@ class SktManager {
...
+  bool DeleteSocketFile(const std::string addr_path);
```
Comment (killight98) [inline src/lava/magma/runtime/message_infrastructure/message_infrastructure/csrc/channel/socket/socket.h:43]:
```
bool DeleteSocketFile(const std::string &addr_path);
```

File: src/lava/magma/runtime/message_infrastructure/message_infrastructure/csrc/channel/socket/socket_port.cc [hunk 3]
```diff
@@ -103,4 +105,105 @@ MetaDataPtr SocketRecvPort::Peek() {
...
+  sfd_ = socket(AF_UNIX, SOCK_STREAM, 0);
+  if (sfd_ == -1) {
+    LAVA_LOG_ERR("Cannot Create Socket Domain File Descripter\n");
+  }
+
+  size_t skt_addr_len = sizeof(sa_family_t) + addr_path_.size();
+  sockaddr *skt_addr = reinterpret_cast<sockaddr*>(malloc(skt_addr_len));
+  skt_addr->sa_family = AF_UNIX;
+  memcpy(&skt_addr->sa_data[0], addr_path.c_str(), addr_path_.size());
+  // printf("the path: %s, %d\n", &skt_addr->sa_data[0], addr_path_.size());
```
Comment (killight98) [inline src/lava/magma/runtime/message_infrastructure/message_infrastructure/csrc/channel/socket/socket_port.cc:164]: remove this line or use debug log.

File: src/lava/magma/runtime/message_infrastructure/message_infrastructure/csrc/core/utils.h [hunk 4]
```diff
@@ -55,7 +55,8 @@ enum ChannelType {
   SHMEMCHANNEL = 0,
   RPCCHANNEL = 1,
   DDSCHANNEL = 2,
-  SOCKETCHANNEL = 3
+  SOCKETCHANNEL = 3,
+  TEMPCHANNEL = 4
```
Comment (killight98) [inline src/lava/magma/runtime/message_infrastructure/message_infrastructure/csrc/core/utils.h:59]: this value is useless?

File: src/lava/magma/runtime/message_infrastructure/message_infrastructure/ports.py [hunk 5]
```diff
@@ -64,3 +65,23 @@ class Channel(CppChannel):
     @property
     def src_port(self):
         return SendPort(super().src_port)
+
+
+class TempChannel(CppTempChannel):
+
+    @property
+    def src_port(self):
+        return SendPort(super().src_port)
```
Comment (killight98) [inline src/lava/magma/runtime/message_infrastructure/message_infrastructure/ports.py:74]: i think we need to check if need to copy source data for TempChannel.

**Review by killight98 — CHANGES_REQUESTED** [review]

File: src/lava/magma/runtime/message_infrastructure/message_infrastructure/csrc/core/utils.h (hunk 4 above)
Comment (hongdami) [reply src/lava/magma/runtime/message_infrastructure/message_infrastructure/csrc/core/utils.h:59]: port_proxy need channel_type.

**Review by killight98 — APPROVED** [review]

**Pull request merged**